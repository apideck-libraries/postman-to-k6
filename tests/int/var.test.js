/* eslint-disable space-before-function-paren */

import convertFile from '../../lib/convert/file';

test('global', async () => {
  const [main] = await convertFile('test/material/2/var-global.json', {
    globals: 'test/material/2/globals.json',
  });
  expect(main).toMatchSnapshot();
});

test('collection', async () => {
  const [main] = await convertFile('test/material/2/var-collection.json');
  expect(main).toMatchSnapshot();
});

test('environment', async () => {
  const [main] = await convertFile('test/material/2/var-environment.json', {
    environment: 'test/material/2/environment.json',
  });
  expect(main).toMatchSnapshot();
});

test('data json', async () => {
  const [main] = await convertFile('test/material/2/var-data-json.json', {
    json: 'test/material/2/data-json.json',
  });
  expect(main).toMatchSnapshot();
});

test('data csv', async () => {
  const [main] = await convertFile('test/material/2/var-data-csv.json', {
    csv: 'test/material/2/data-csv.csv',
  });
  expect(main).toMatchSnapshot();
});

test('data custom iterations', async () => {
  const [main] = await convertFile('test/material/2/var-data-json.json', {
    json: 'test/material/2/data-json.json',
    iterations: 27,
  });
  expect(main).toMatchSnapshot();
});

test('address start', async () => {
  const [main] = await convertFile('test/material/2/var-address-start.json');
  expect(main).toMatchSnapshot();
});
