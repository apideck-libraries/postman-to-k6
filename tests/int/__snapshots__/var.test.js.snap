// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`address start 1`] = `
"// Auto-generated by the postman-to-k6 converter

import "./libs/shim/core.js";
import "./libs/shim/urijs.js";

export let options = { maxRedirects: 4 };

const Request = Symbol.for("request");
postman[Symbol.for("initial")]({
  options,
  collection: {
    domain: "example.com",
    machine: 573
  }
});

export default function() {
  postman[Request]({
    name: "TestRequest",
    method: "GET",
    address: "{{machine}}.{{domain}}"
  });
}
"
`;

exports[`collection 1`] = `
"// Auto-generated by the postman-to-k6 converter

import "./libs/shim/core.js";

export let options = { maxRedirects: 4 };

const Request = Symbol.for("request");
postman[Symbol.for("initial")]({
  options,
  collection: {
    domain: "example.com",
    machine: 573
  }
});

export default function() {
  postman[Request]({
    name: "TestRequest",
    method: "GET",
    address: "http://{{machine}}.{{domain}}/"
  });
}
"
`;

exports[`data csv 1`] = `
"// Auto-generated by the postman-to-k6 converter

import "./libs/shim/core.js";
import papaparse from "./libs/papaparse.js";

export let options = { maxRedirects: 4 };

const file = (() => {
  // Load data file
  const text = open("./data.csv");
  const rows = papaparse.parse(text, { header: true }).data;
  return rows;
})();
options.iterations = file.length;

const Iteration = Symbol.for("iteration");
const Request = Symbol.for("request");
postman[Symbol.for("initial")]({
  options,
  data: file
});

export default function() {
  postman[Iteration]();

  postman[Request]({
    name: "TestRequest",
    method: "GET",
    address: "http://{{first}}.{{third}}/"
  });
}
"
`;

exports[`data custom iterations 1`] = `
"// Auto-generated by the postman-to-k6 converter

import "./libs/shim/core.js";

export let options = { maxRedirects: 4, iterations: 27 };

const file = (() => {
  // Load data file
  const text = open("./data.json");
  const rows = JSON.parse(text);
  return rows;
})();

const Iteration = Symbol.for("iteration");
const Request = Symbol.for("request");
postman[Symbol.for("initial")]({
  options,
  data: file
});

export default function() {
  postman[Iteration]();

  postman[Request]({
    name: "TestRequest",
    method: "GET",
    address: "http://{{first}}.{{third}}/"
  });
}
"
`;

exports[`data json 1`] = `
"// Auto-generated by the postman-to-k6 converter

import "./libs/shim/core.js";

export let options = { maxRedirects: 4 };

const file = (() => {
  // Load data file
  const text = open("./data.json");
  const rows = JSON.parse(text);
  return rows;
})();
options.iterations = file.length;

const Iteration = Symbol.for("iteration");
const Request = Symbol.for("request");
postman[Symbol.for("initial")]({
  options,
  data: file
});

export default function() {
  postman[Iteration]();

  postman[Request]({
    name: "TestRequest",
    method: "GET",
    address: "http://{{first}}.{{third}}/"
  });
}
"
`;

exports[`environment 1`] = `
"// Auto-generated by the postman-to-k6 converter

import "./libs/shim/core.js";

export let options = { maxRedirects: 4 };

const Request = Symbol.for("request");
postman[Symbol.for("initial")]({
  options,
  environment: {
    first: "one",
    second: "two",
    third: "three"
  }
});

export default function() {
  postman[Request]({
    name: "TestRequest",
    method: "GET",
    address: "http://{{first}}.{{third}}/"
  });
}
"
`;

exports[`global 1`] = `
"// Auto-generated by the postman-to-k6 converter

import "./libs/shim/core.js";

export let options = { maxRedirects: 4 };

const Request = Symbol.for("request");
postman[Symbol.for("initial")]({
  options,
  global: {
    first: "one",
    second: "two",
    third: "three"
  }
});

export default function() {
  postman[Request]({
    name: "TestRequest",
    method: "GET",
    address: "http://{{first}}.{{third}}/"
  });
}
"
`;
